lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -3098,6 +3098,7 @@ defmodule LightningWeb.WorkflowLive.Edit do
     |> assign(show_workflow_ai_chat: show_workflow_ai_chat)
     |> assign_changeset(changeset)
     |> maybe_disable_canvas()
+    |> generate_workflow_code()
```

Comment:
I think we already have a call to `generate_workflow_code` that could deal with this issue. 

**Check below**
https://github.com/OpenFn/lightning/blob/b7522e28a606cb5a456a6619064d3ffeaa43522a/lib/lightning_web/live/workflow_live/edit.ex#L3284-L3294

In this code, we're restricting the function to only run when there's a diff between two states of the workflow. Hence for a new workflow, there will be no patches available and then the function wouldn't be called. Is there another reason we put the function in there? if not, I believe removing the existing call from that if-block should resolve this issue without introducing another call to the function.

Reply:
Hey @elias-ba, sorry I missed the notification for your response. 

Doing a follow-up PR is fine, but I was hoping to get clarity as why we can't move the `generate_workflow_code()` out of the `if do` block as Farhan suggested above. Calling `generate_workflow_code()` inside `assign_workflow` means it will be called twice if there is a follow-up call to `push_patches_applied/2`

Reply:
Had a chat in slack, `generate_workflow_code` is harmless and so there is no problem calling it more than once